client/src/components/Courses.tsx

```diff
@@ -23,26 +40,103 @@ const CourseList = styled.div`
   height: 250px;
   border: 1px solid #c4c4c4;
   border-radius: 10px;
-  flex: 3;
+  flex: 1.4;
 `;
 
-function Courses() {
+interface CoursesProps {
+  courseSelected?: string[];
+  addCourseToRedux?: any;
+}
+
+function Courses({ courseSelected, addCourseToRedux }: CoursesProps) {
+  const classes = useStyles();
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [coursesPool, setCoursesPool] = useState<string[]>([]);
+  const [currCoursesSelected, setCurrCoursesSelected] = useState<string[]>(
+    courseSelected ? courseSelected : []
+  );
+  const [curr, setCurr] = useState("");
+
+  const fetchAllCourses = async () => {
+    const coursesFetched = await axios.get("http://localhost:3000/courses");
```

Comment:
Just a thought but I think it'd be beneficial to create a separate config file to store the base urls which will be environment dependent so that if/when we choose to deploy somewhere, we can just use the correct env and that'll reference the correct base url.

Reply:
@jacksonyuanjx Good point Jackson. But for now it only receives a mock response to demonstrate that this auto-complete function works at a minimal level. We can discuss how we gonna manage all APIS used by FE in a new issue or in our next meeting on Saturday.